common/djangoapps/third_party_auth/management/commands/tests/test_saml.py

```diff
@@ -416,57 +392,122 @@ def test_run_checks_slug_mismatches(self, mock_set_custom_attribute):
             entity_id='https://example.com'
         )
 
-        SAMLProviderConfigFactory.create(
+        provider = SAMLProviderConfigFactory.create(
             site=self.site,
             slug='provider-slug',
             saml_configuration=config
         )
 
         output = self._run_checks_command()
 
-        self.assertIn('[WARNING]', output)
-        self.assertIn('provider-slug', output)
-        self.assertIn('does not match the provider\'s slug', output)
-
-        # Check observability calls
-        expected_calls = [
-            mock.call('saml_management_command.operation', 'run_checks'),
-            mock.call('saml_management_command.total_providers', 2),
-            mock.call('saml_management_command.outdated_count', 0),
-            mock.call('saml_management_command.site_mismatch_count', 0),
-            mock.call('saml_management_command.slug_mismatch_count', 1),
-            mock.call('saml_management_command.null_config_count', 1),
-            mock.call('saml_management_command.error_count', 0),
-            mock.call('saml_management_command.total_requiring_attention', 1),
-        ]
-        mock_set_custom_attribute.assert_has_calls(expected_calls, any_order=False)
-
-    @mock.patch('common.djangoapps.third_party_auth.management.commands.saml.set_custom_attribute')
-    def test_run_checks_null_configurations(self, mock_set_custom_attribute):
+        expected_info = (
+            f'[INFO] Provider (id={provider.id}, name={provider.name}, '
+            f'slug={provider.slug}, site_id={provider.site_id}) '
+            f'has SAML config (id={config.id}, slug=\'{config.slug}\') '
+            f'that does not match the provider\'s slug.'
+        )
+        self.assertIn(expected_info, output)
+        self.assertIn('Slug mismatches: 1', output)
+
+    def test_run_checks_null_configurations(self):
         """
         Test the --run-checks command identifies providers with null configurations.
         """
-        SAMLProviderConfigFactory.create(
+        provider = SAMLProviderConfigFactory.create(
             site=self.site,
             slug='null-provider',
             saml_configuration=None
         )
 
+        with mock.patch('common.djangoapps.third_party_auth.models.SAMLConfiguration.current',
+                        return_value=None):
+            output = self._run_checks_command()
+
+        expected_warning = (
+            f'[WARNING] Provider (id={provider.id}, name={provider.name}, '
+            f'slug={provider.slug}, site_id={provider.site_id}) '
+            f'has no direct SAML configuration and no matching default configuration was found.'
+        )
+        self.assertIn(expected_warning, output)
+        self.assertIn('Missing configs: 2', output)
```

Comment:
It may be useful to add (or restore) a quick comment about why this is 2 and not 1.